cnd/src/network/protocol.rs

```diff
@@ -0,0 +1,207 @@
+use crate::swap_protocols::{
+    rfc003::{Secret, SecretHash},
+    SwapId,
+};
+use futures::{future, AsyncRead, AsyncWrite, Future};
+use libp2p::core::{upgrade, InboundUpgrade, OutboundUpgrade, UpgradeInfo};
+use serde::{Deserialize, Serialize};
+use std::{fmt, io, iter, pin::Pin, str::FromStr};
+use tracing::{debug, trace};
+
+/// The secret hash sharing protocol works in the following way:
+///
+/// - Dialer (Alice) writes the `SecretHashInfo` to the substream.
+/// - Listener (Bob) reads the `SecretHashInfo` from the substream.
+
+/// Represents a prototype for an upgrade to handle the secret hash sharing
+/// protocol.
+#[derive(Debug, Clone)]
+pub struct SecretHashProtocolConfig;
+
+/// Data sent to peer in secret hash protocol.
+#[derive(Deserialize, Debug, Serialize)]
+pub struct SecretHashInfo {
+    swap_id: SwapId,
+    secret_hash: SecretHash,
+}
+
+lazy_static::lazy_static! {
```

Comment:
The sizes are always the same. Just add a test if you want to be sure, no need to pay this runtime overhead.